tests/numeric_test.py

```diff
@@ -197,21 +197,341 @@ def testCumProd(self):
             ak.cumprod([range(0, 10)])
 
     def testSin(self):
-        na = np.linspace(1, 10, 10)
+        na = np.arange(1, 10, dtype="int64")
         pda = ak.array(na)
+        self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
+
+        na = np.arange(1, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
 
+        na = np.arange(1, 10, dtype="float64")
+        pda = ak.array(na)
         self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
+
         with self.assertRaises(TypeError):
-            ak.cos([range(0, 10)])
+            ak.sin([range(0, 10)])
 
     def testCos(self):
-        na = np.linspace(1, 10, 10)
+        na = np.arange(1, 10, dtype="int64")
         pda = ak.array(na)
+        self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
 
+        na = np.arange(1, 10, dtype="uint64")
+        pda = ak.array(na)
         self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
+
+        na = np.arange(1, 10, dtype="float64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.cos(na), ak.cos(pda).to_ndarray()))
+
         with self.assertRaises(TypeError):
             ak.cos([range(0, 10)])
 
+    def testTan(self):
+        na = np.arange(1, 10, dtype="int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        na = np.arange(1, 10, dtype="uint64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        na = np.arange(1, 10, dtype="float64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.tan(na), ak.tan(pda).to_ndarray()))
+
+        with self.assertRaises(TypeError):
+            ak.tan([range(0, 10)])
+
+    def testArcsin(self):
+        na = np.arange(0, 1, 0.1, dtype="int64")
```

Comment:
i don't feel like doing `arange` with a step of `0.1` is really what we want here
```python
In [2]: na = np.arange(0, 1, 0.1, dtype="int64")

In [3]: na
Out[3]: array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
```

Reply:
Updated